isis/tests/SpiceRotationTests.cpp

```diff
@@ -0,0 +1,822 @@
+  EXPECT_NEAR(sysNutPrec0[5].degrees(), 229.8, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[6].degrees(), 352.25, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[7].degrees(), 113.35, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[8].degrees(), 146.64, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[9].degrees(), 49.24, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[10].degrees(), 99.360714, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[11].degrees(), 175.895369, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[12].degrees(), 300.323162, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[13].degrees(), 114.012305, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[14].degrees(), 49.511251, testTolerance);
+
+  vector<Angle> sysNutPrec1 = ioRot.sysNutPrecCoefs();
+  EXPECT_EQ(sysNutPrec1.size(), 15);
+  EXPECT_NEAR(sysNutPrec1[0].degrees(), 91472.9, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[1].degrees(), 45137.2, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[2].degrees(), 4850.7, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[3].degrees(), 1191.3, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[4].degrees(), 262.1, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[5].degrees(), 64.3, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[6].degrees(), 2382.6, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[7].degrees(), 6070, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[8].degrees(), 182945.8, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[9].degrees(), 90274.4, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[10].degrees(), 4850.4046, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[11].degrees(), 1191.9605, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[12].degrees(), 262.5475, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[13].degrees(), 6070.2476, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[14].degrees(), 64.3, testTolerance);
+
+  // These are doubles so we can use vector comparison
+  EXPECT_PRED_FORMAT3(AssertVectorsNear, ioRot.poleRaNutPrecCoefs(),
+                      (vector<double>{0.0, 0.0, 0.094, 0.024, 0.0,
+                                      0.0, 0.0, 0.0, 0.0, 0.0,
+                                      0.0, 0.0, 0.0, 0.0, 0.0}),
+                      testTolerance);
+  EXPECT_PRED_FORMAT3(AssertVectorsNear, ioRot.poleDecNutPrecCoefs(),
+                      (vector<double>{0.0, 0.0, 0.04, 0.011, 0.0,
+                                      0.0, 0.0, 0.0, 0.0, 0.0,
+                                      0.0, 0.0, 0.0, 0.0, 0.0}),
+                      testTolerance);
+  EXPECT_PRED_FORMAT3(AssertVectorsNear, ioRot.pmNutPrecCoefs(),
+                      (vector<double>{0.0, 0.0, -0.085, -0.022, 0.0,
+                                      0.0, 0.0, 0.0, 0.0, 0.0,
+                                      0.0, 0.0, 0.0, 0.0, 0.0}),
+                      testTolerance);
+}
+
+
+TEST_F(SpiceRotationIsd, FromALE) {
+  // Test with just a time dependent rotation
+  SpiceRotation aleQuatRot(-94031);
+  aleQuatRot.LoadCache(isd);
+
+  EXPECT_EQ(aleQuatRot.getFrameType(), SpiceRotation::CK);
+  EXPECT_TRUE(aleQuatRot.IsCached());
```

Comment:
If this failed, would it cause any unwanted errors in the checks below?

Reply:
If this is false, then I would not expect unusual failures in the code after it.